### PR TITLE
Rename IRequest related variable and method names

### PR DESCRIPTION
**Pull request by joeldodge79**
Make the logic be more about existing language features than
needing to introduce crutches

### CODE REVIEW

**Comment by joeldodge79** [comment]
As I was adding `useNamedArguments` for python makeTheCall I realized that I like this gesture better. But happy to discuss if you disagree

**Review by jkaster — DISMISSED** [review]
Good rename. It provides better clarity for what exactly that generation toggle means.

File: packages/sdk-codegen/src/codeGen.ts [hunk 1]
```diff
@@ -622,7 +623,7 @@ export interface ICodeGen {
 }
 
 export abstract class CodeGen implements ICodeGen {
-  needsRequestTypes = false
+  useNamedParameters = false
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/codeGen.ts:626]: whoops, this should default true in keeping with prior default
```suggestion
  useNamedParameters = true
```

File: packages/sdk-codegen/src/typescript.gen.ts [hunk 2]
```diff
@@ -62,7 +62,6 @@ export class TypescriptGen extends CodeGen {
 
   indentStr = '  '
   endTypeStr = '\n}'
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/typescript.gen.ts:64]:
```suggestion
  endTypeStr = '\n}'
  useNamedParameters = false
```

File: packages/sdk-codegen/src/csharp.gen.ts [hunk 3]
```diff
@@ -137,7 +137,7 @@ export class CSharpGen extends CodeGen {
 
   indentStr = '  '
   endTypeStr = '\n}'
-  needsRequestTypes = false
+  useNamedParameters = true
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/csharp.gen.ts:140]:
```suggestion
```

File: packages/sdk-codegen/src/kotlin.gen.ts [hunk 4]
```diff
@@ -55,7 +55,7 @@ export class KotlinGen extends CodeGen {
 
   indentStr = '  '
   endTypeStr = '\n) : Serializable'
-  needsRequestTypes = false
+  useNamedParameters = true
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/kotlin.gen.ts:58]:
```suggestion
```

File: packages/sdk-codegen/src/python.gen.ts [hunk 5]
```diff
@@ -50,6 +50,7 @@ export class PythonGen extends CodeGen {
   dataStructureDelimiter = ', '
   enumDelimiter = '\n'
   codeQuote = '"'
+  useNamedParameters = true
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/python.gen.ts:53]:
```suggestion
```

File: packages/sdk-codegen/src/swift.gen.ts [hunk 6]
```diff
@@ -55,7 +55,7 @@ export class SwiftGen extends CodeGen {
 
   indentStr = '    '
   endTypeStr = `\n}`
-  needsRequestTypes = false
+  useNamedParameters = true
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/swift.gen.ts:58]:
```suggestion
```

**Review by joeldodge79 — COMMENTED** [review]
@jkaster I goofed on the default value for `useNamedParameters`. I updated it to `true` overrode to `false` for typescript, and removed unnecessary declaration in the other languages.

If that looks ok I'll commit these and need another review. Otherwise, if you'd prefer the default for `useNamedParameters` to be `false` then I'll leave it as is (but I think more languages support it than not in general)

**Review by josephaxisa — APPROVED** [review]
This all looks ok to me

**Pull request merged**